cmd/crdvalidator/handlers/crd.go

```diff
@@ -49,16 +51,18 @@ func NewCrdValidator(log logr.Logger, client client.Client) CrdValidator {
 // a safe upgrade based on the crd.Validate() function call
 func (cv *CrdValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
 	incomingCrd := &apiextensionsv1.CustomResourceDefinition{}
-
-	err := cv.decoder.Decode(req, incomingCrd)
-	if err != nil {
+	if err := cv.decoder.Decode(req, incomingCrd); err != nil {
 		message := fmt.Sprintf("failed to decode CRD %q", req.Name)
 		cv.log.V(0).Error(err, message)
 		return admission.Errored(http.StatusBadRequest, fmt.Errorf("%s: %w", message, err))
 	}
 
-	err = crd.Validate(ctx, cv.client, incomingCrd)
-	if err != nil {
+	// Check if the request should get validated
+	if !enabled(incomingCrd) {
+		return admission.Allowed("")
+	}
+
+	if err := crd.Validate(ctx, cv.client, incomingCrd); err != nil {
 		message := fmt.Sprintf("failed to validate safety of %s for CRD %q: %s", req.Operation, req.Name, err)
 		cv.log.V(0).Info(message)
 		return admission.Denied(message)
```

Comment:
Since this webhook is operating on a built-in type (where no upstream documentation will exist about this webhook), should we include something in the deny message that mentions the annotation that's used? That way a user can know what to change to get a different answer from this webhook.

Reply:
Sure, that sounds fair.